modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/refresh/RefreshPodcastsThread.kt

```diff
@@ -270,6 +273,11 @@ class RefreshPodcastsThread(
         getEntryPoint().settings().setRefreshState(RefreshState.Failed(message))
     }
 
+    private fun dispatchCurrentRefreshedState() {
+        val settings = getEntryPoint().settings()
+        (settings.getLastSuccessRefreshState() ?: RefreshState.Never).let(settings::setRefreshState)
```

Comment:
There is an edge case where we lose failure state due to this but I don't see any way around it without rewriting a lot of code. I think this is an acceptable compromise.